src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -139,5 +162,34 @@ public TemplateCache(JObject? contentJobject, ILogger logger)
             while (currentCulture.Name != CultureInfo.InvariantCulture.Name);
             return null;
         }
+
+        // add warning for the case when there is an attempt to overwrite existing managed by new managed template
+        private void PrintOverlappingIdentityWarning(IDictionary<string, IList<(ITemplate Template, string PackageDisplayName, ILocalizationLocator? Localization)>> templateDeduplicationDictionary)
+        {
+            foreach (var identityToTemplates in templateDeduplicationDictionary)
+            {
+                // we print the message only if managed template wins and we have > 1 managed templates with overlapping identities
+                var lastTemplate = identityToTemplates.Value.LastOrDefault();
```

Comment:
```suggestion
                var lastTemplate = identityToTemplates.Value.Last();
```

Reply:
if last template is not IManagedTemplatePackage, we can continue and avoid creating the array.